### ISSUE TITLE
Share to Slack

### ISSUE THREAD

**Issue by EdwardTeach68** [yasirkula/UnityNativeShare#92]
Hi there.
After sharing image to slack it doesn't hide share dialogue.ShareResult also NotShared,but image was shared.How can i hide share dialogue?

**Comment by yasirkula** [comment]
Are there any error messages in Xcode console?

**Comment by EdwardTeach68** [comment]
I don't see any errors in the console.

**Comment by EdwardTeach68** [comment]
The app becomes non-clickable with black bars around the edges. Swiping down makes it clickable again and full screen.This issue with slack only

**Comment by yasirkula** [comment]
Can you see the "Shared to %@ with result: %d" log in Xcode console for Slack or any other apps? If so, does this log appear before or after the share result log that is Debug.Log'ed from Unity?

**Comment by EdwardTeach68** [comment]
3dAssistant[14888:3781953] Shared to com.tinyspeck.chatlyio.share with result: 0
NotShared
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Data.Scripts.ScreenShots.<>c:<TakeScreenshot>b__2_0(ShareResult, String)
ShareResultCallback:Invoke(ShareResult, String)
NativeShareNamespace.NSShareResultCallbackiOS:OnShareCompleted(String)
 
(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
Ok,i see that after slack sharing my app is not focused,only after swiping down it showes this debug.Looks like slack does not exit from his sharing dialogue and a swipe down closes it

**Comment by yasirkula** [comment]
Result of 0 should actually point to ShareResult.Unknown. I'm not sure why it resolves to NotShared in your case. Can you insert `Debug.Log("Share message received: " + message);` to [this line](https://github.com/yasirkula/UnityNativeShare/blob/a325dd8ce0fa137d55994c11ad74267560f13e6b/Plugins/NativeShare/iOS/NSShareResultCallbackiOS.cs#L28) and see what it outputs?

**Comment by EdwardTeach68** [comment]
Yeah,its slack problem.I have shared an image from Telegram to Slack and get the same issue.Thank you for helping)

**Comment by yasirkula** [comment]
I'm still unsure about the cause of NotShared permission. It should've been Unknown. Did you get a chance to insert the Debug.Log line I've mentioned to the code?

**Comment by EdwardTeach68** [comment]
Ok,one moment

**Comment by EdwardTeach68** [comment]
Share message received: 2com.tinyspeck.chatlyio.share (show first)
after shows debug.log from callback "NotShared"
and Share message received: 2(null)

**Comment by yasirkula** [comment]
Ah, I completely forgot about the `completed ? 1 : 2` line [here](https://github.com/yasirkula/UnityNativeShare/blob/a325dd8ce0fa137d55994c11ad74267560f13e6b/Plugins/NativeShare/iOS/NativeShare.mm#L40). So the issue is that Slack is returning _false_ as `BOOL completed` parameter. I haven't seen other apps having this issue, though. If you are rewarding the player in ShareResultCallback, you can query `result != NativeShare.ShareResult.NotShared || !string.IsNullOrEmpty(shareTarget)` for maximum compatibility. This way, you'll make sure that the user has at least selected an app from the share sheet.
